### ISSUE TITLE
[bug] Exception in full_deploy if package has symlinks and destination folder is on a Windows share

### ISSUE THREAD

**Issue by fschoenm** [conan-io/conan#14451]
### Environment details

* Operating System+version: Ubuntu 23.04
* Compiler+version: n/a
* Conan version: 2.0.9
* Python version: 3.11.4

### Steps to reproduce

Rare issue but if on Linux you run `full_deploy` onto a Windows network share, packages with symlinks will fail the whole deployment process. Try e.g. with _fast-dds_ package that contains a symlink:
```
$ conan install --requires 'fast-dds/[]' -s build_type=Debug --deployer=full_deploy
```
These are the mount properties:
```
...  type cifs (rw,relatime,vers=3.0,cache=strict,username=user,domain=domain,uid=1001,noforceuid,gid=123,noforcegid,addr=aa.bb.cc.dd,file_mode=0664,dir_mode=0775,soft,nounix,serverino,mapposix,rsize=65536,wsize=65536,bsize=1048576,echo_interval=60,actimeo=1,closetimeo=1)
```

Maybe there's a way to handle this scenario or call `copytree()` with different parameters to work in this situation.

### Logs

```
...
======== Finalizing install (deploy, generators) ========
cli: Conan built-in full deployer to /mnt/nu3008/quantum/temp/fschoenm/conan-test
Traceback (most recent call last):
  File "/home/fschoenm/.local/pipx/venvs/conan/lib/python3.11/site-packages/conan/cli/cli.py", line 272, in main
    cli.run(args)
  File "/home/fschoenm/.local/pipx/venvs/conan/lib/python3.11/site-packages/conan/cli/cli.py", line 172, in run
    command.run(self._conan_api, args[0][1:])
  File "/home/fschoenm/.local/pipx/venvs/conan/lib/python3.11/site-packages/conan/cli/command.py", line 125, in run
    info = self._method(conan_api, parser, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fschoenm/.local/pipx/venvs/conan/lib/python3.11/site-packages/conan/cli/commands/install.py", line 84, in install
    conan_api.install.install_consumer(deps_graph=deps_graph,
  File "/home/fschoenm/.local/pipx/venvs/conan/lib/python3.11/site-packages/conan/api/subapi/install.py", line 66, in install_consumer
    do_deploys(self.conan_api, deps_graph, deploy, base_folder)
  File "/home/fschoenm/.local/pipx/venvs/conan/lib/python3.11/site-packages/conan/internal/deploy.py", line 45, in do_deploys
    deployer(graph=graph, output_folder=deploy_folder)
  File "/home/fschoenm/.local/pipx/venvs/conan/lib/python3.11/site-packages/conan/internal/deploy.py", line 71, in full_deploy
    shutil.copytree(dep.package_folder, new_folder, symlinks=True)
  File "/usr/lib/python3.11/shutil.py", line 561, in copytree
    return _copytree(entries=entries, src=src, dst=dst, symlinks=symlinks,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/shutil.py", line 515, in _copytree
    raise Error(errors)
shutil.Error: [('/home/fschoenm/.conan2/p/fast-811fbe8135a6e/p/bin/fast-discovery-server', '/mnt/nu3008/quantum/temp/fschoenm/conan-test/full_deploy/host/fast-dds/2.10.1/Debug/x86_64/bin/fast-discovery-server', "[Errno 95] Operation not supported: 'fast-discovery-serverd-1.0.1' -> '/mnt/nu3008/quantum/temp/fschoenm/conan-test/full_deploy/host/fast-dds/2.10.1/Debug/x86_64/bin/fast-discovery-server'")]
```

**Comment by memsharded** [comment]
Hi @fschoenm 

I have been checking this and seems complicated. The issue is happening in the Python internals of ``shutil.copytree`` and it seems there are no other parameters or behavior that could be changed in this call that would affect. It seems a limitation in the Python standard library, maybe it would make sense to report it there. Any Python experts that could help here? (marking this as help wanted)


I'd recommend to try to deploy locally to the local filesystem, then do a system copy to the Windows share. Is a normal ``cp -R`` working in that case and moves the symlinks correctly to the Windows share?

**Comment by fschoenm** [comment]
@memsharded I found the following ticket referencing this issue: python/cpython#73702, so at least this seems to be a known issue.

I can think of the following solutions:
* Don't set `symlinks=True` (or try again with `symlinks=False` in case of an error), which would not create symlinks but create a copy of the file in question.
* Catch the error and try to handle it "somehow", e.g. show a warning to the user that a file could not be copied but at least don't throw an exception and stop the whole install/deploymeny process.
* Allow a command line option or configuration setting to make `symlinks=True/False` configurable.

I don't know if you prefer any of those solutions but there doesn't seem a good way to handle this. Some deployment options simply don't support symlinks.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/issues/14456 suggest that adding developer mode to the Windows machine can solve some issues, maybe checking if it is enabled in the OS of the Windows network share?

**Comment by fschoenm** [comment]
@memsharded  Unfortunately I cannot try (or request to enable) this because the share is a centrally managed network share on a server.

**Comment by memsharded** [comment]
Proposing the ``conf`` approach in https://github.com/conan-io/conan/pull/14461, if you can run it from source, feedback is welcome

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/14461 has already been merged for 2.0.10, but if possible, early testing and feedback would be great.
